### ISSUE TITLE
Guidance for DCO (and DCO vs. CLA)

### ISSUE THREAD

**Issue by tobie** [openjs-foundation/cross-project-council#387]
In https://github.com/openjs-foundation/cross-project-council/issues/124#issuecomment-512450068, @brianwarner announced that the foundation was working with its IP counsel to define the best way to offer a CLA for projects which want to continue using one and was planning to publish guidance on this topic. That looks super useful. ❤️

Is there a similar plan for projects which want to use a DCO instead? If so, what is the timeframe for guidance to be available to projects and what tooling, if any, is planned?

Also, is there a plan to release guidance for projects as to which one of CLA or DCO they should pick and why?

**Comment by mhdawson** [comment]
@brianwarner any starting comments/info?

**Comment by tobie** [comment]
I read through the [IP Policy](https://ip-policy.openjsf.org) and while it clarifies clearly the relationship between the DCO and the CLA, and the requirement to at least require a DCO, it doesn't say anything about the actual implementation of the policy.

Is that left to the project to decide?

If so, is it enough to include the text of the [DCO](https://developercertificate.org/) in CONTRIBUTING.md, the [way node.js does](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)?

That would be ideal to lower the friction for contributing to the project.

**Comment by Trott** [comment]
> If so, is it enough to include the text of the [DCO](https://developercertificate.org/) in CONTRIBUTING.md, the way [node.js does](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)?

Node.js also [includes the DCO text in the pull request template](https://raw.githubusercontent.com/nodejs/node/115792dfde436354c82f38361cbff75ad226b8d0/.github/PULL_REQUEST_TEMPLATE.md) if that makes a difference.

**Comment by tobie** [comment]
> Node.js also includes the DCO text in the pull request template if that makes a difference.

Oh! As a comment! That's pretty smart!

**Comment by kborchers** [comment]
I think @brianwarner and @mkdolan can provide more official guidance here but my understanding is that the Node.js implementation of DCO is still insufficient and that use of the DCO will require appropriate `Signed-off by ...` statements in commit messages in order to be accepted.

**Comment by tobie** [comment]
> use of the DCO will require appropriate `Signed-off by ...` statements in commit messages in order to be accepted.

That’s precisely the part that is difficult to implement for contributors and that the AMP TSC would absolutely want to avoid.

**Comment by ljharb** [comment]
Don't forget that https://ben.balter.com/2018/01/02/why-you-probably-shouldnt-add-a-cla-to-your-open-source-project/ is also a pretty good option.

**Comment by tobie** [comment]
I agree, @ljharb. I think @davidstrauss astutely frames the issue in https://github.com/ampproject/meta-tsc/issues/25#issuecomment-549540646: although more lightweight on the legal requirements it imposes on contributors, DCO _implementations_ can end-up adding _more_ friction to contributions than a well implemented CLA. I’d like us to avoid this pitfall.

**Comment by ljharb** [comment]
In https://github.com/openjs-foundation/cross-project-council/pull/386#discussion_r353395129 @brianwarner points out that the node foundation didn't require a CLA; did it require a DCO? If not, then why would OpenJS projects require a DCO in the absence of a CLA?

**Comment by brianwarner** [comment]
@ljharb Node Foundation did require a DCO, yes.

**Comment by mcollina** [comment]
The Node.js project never enforced the letter of the DCO, as there is no signoff in any most of the commits.

**Comment by christian-bromann** [comment]
Has there been any updated to this? I found a [comment](https://github.com/ampproject/meta-tsc/issues/25#issuecomment-549510504) from @MylesBorins saying that the foundation is finalizing the language of the
CLA.

**Comment by tobie** [comment]
Yes. The OpenJSF's board has recently approved a CLA for [individual](https://individual-cla.openjsf.org/) and [corporate](https://corporate-cla.openjsf.org/) contributors and is relying on the Linux Foundation's [EasyCLA tool](https://lfcla.com/) to implement it. More info [here](https://openjsf.org/about/the-openjs-foundation-cla/).

**Comment by christian-bromann** [comment]
Thanks @tobie , I will sign up for EasyCLA and will look for a place to document the process as guidance for other projects.

**Comment by tobie** [comment]
@brianwarner might be able to help set you up.

**Comment by brianwarner** [comment]
I can, yes. 

@christian-bromann let's talk, I can walk you through the process.

**Comment by tobie** [comment]
There's also a test repo, if you want to look at what the outcome looks like: https://github.com/openjs-foundation/TEST-EasyCLA

**Comment by bnb** [comment]
cc @lukeschantz would be great to figure out a resolution to this super old issue :)

**Comment by tobie** [comment]
Fixed in #1354.
